src/Button/index.jsx

```diff
@@ -4,40 +4,83 @@ import PropTypes from 'prop-types';
 
 import styles from './Button.scss';
 
-function Button(props) {
-  const {
-    buttonType,
-    className,
-    label,
-    inputRef,
-    isClose,
-    onBlur,
-    onClick,
-    onKeyDown,
-    type,
-    ...other
-  } = props;
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <button
-      className={classNames([
-        ...className,
-        styles.btn,
-      ], {
-        [styles[`btn-${buttonType}`]]: buttonType !== undefined,
-      }, {
-        [styles.close]: isClose,
-      })}
-      onBlur={onBlur}
-      onClick={onClick}
-      onKeyDown={onKeyDown}
-      type={type}
-      ref={inputRef}
-      {...other}
-    >
-      {label}
-    </button>
-  );
+    const {
+      onBlur,
+      onKeyDown,
+    } = props;
+
+    this.onBlur = onBlur.bind(this);
+    this.onKeyDown = onKeyDown.bind(this);
+    this.onClick = this.onClick.bind(this);
+    this.setRefs = this.setRefs.bind(this);
+  }
+
+  /*
+    The button component is given focus explicitly in its onClick to account
+    for the fact that an HTML <button> element in Firefox and Safari does not get
+    focus on onClick.
+
+    See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button.
+  */
+  onClick(e) {
+    this.buttonRef.focus();
+    this.props.onClick(e);
+  }
+
+  /*
+    The button component needs a ref to itself to be able to force
+    focus in its onClick function (buttonRef). It also needs to accept
+    a callback function from parent components to give those parents
+    a reference to their child button (e.g. for the modal component).
+    Therefore, both have been wrapped in a function bound on the class,
+    since one cannot set two ref attributes on a component.
+  */
+  setRefs(input) {
+    this.buttonRef = input;
+    this.props.inputRef(input);
+  }
+
+  render() {
+    const {
+      buttonType,
+      className,
+      label,
+      isClose,
+      type,
+      /* inputRef is not used directly in the render, but it needs to be assigned
+        here to prevent it from being passed to the HTML button component as part
+        of other.
+      */
+      inputRef,
```

Comment:
I don't love that I have to do this here... I wanted to remove this destructuring because props that are not used immediately in the render but by other auxiliary functions get passed to the HTML button component as part of other. For example, inputRef is now used in setRefs and not in the render, so to prevent it from being passed explicitly to the HTML button, I need to be explicitly assign it here so that is not part of other, which is conceptually kind of confusing, since it's not being used in the render. If I remove the destructuring, however, I remove the ability to pass other attributes to the HTML button that are not explicit props of the Button. Any insight/recommendation about how to proceed?

Reply:
This is the same thing I did for [`RadioButtons`](https://github.com/edx/paragon/blob/master/src/RadioButtonGroup/index.jsx#L103-L114)...I'm not sure there's a better solution.

![alt-text](https://media.giphy.com/media/XeXzWgD6P4LG8/giphy.gif)

Reply:
Oh, well! Thanks for taking a look!